feat(authentication): Add information about account linking (#376)
clerkinc/clerk-docs: docs/authentication/social-connections/account-linking.mdx

---
title: Account Linking
description: Learn how Clerk handles account linking and manages unverified email addresses from OAuth providers.
---

# Account Linking

Account Linking is a process used by Clerk to ensure a smooth sign-in and sign-up experience using both OAuth and other methods (e.g. username/password). By using the "email address" as the common identifier, Clerk aims to merge accounts whenever possible. This mechanism is triggered when an OAuth provider returns an email address that matches an existing account, assuming a single owner for each email address.

## How it works

> When a user attempts to sign-in or sign-up, Clerk validates the email address provided. If the email comes from an OAuth provider, there are several scenarios depending on the verification status of the email and whether an account with the same email already exists. Below are the different scenarios and how Clerk handles them.

Reviewer: :question: Should the verbs be "sign in" and "sign up"? Without the hyphen.

Suggestion
```
When a user attempts to sign in or sign up, Clerk checks the provided email address. If the email address comes from an OAuth provider, depending on the email address verification status and whether an account with the same email address exists, Clerk will try to link the email address with the existing account. 

Let's look at the different possibilities and how Clerk handles each of them.
```

Author: ✅ Added a bit more context, plus your suggestions

<Images
    width={3333}
    height={3551}
    src="/docs/images/authentication/account-linking-flow.webp"
    alt="Flow chart of the account linking process in various scenarios"
/>

### When email address is verified in both OAuth and Clerk

> In this scenario, when a user with a verified email address signs in using an OAuth provider that returns a matching verified email, Clerk links the OAuth account to the existing account and signs the user in. This is valid even if the existing account has a password.

Reviewer: :upside_down_face: Let's expand on the "even if has a password" case? Explain that the user will be signed in skipping the password check.

Author: ✅ Changed the sentence a bit.

### When email address is verified in Clerk but not in OAuth

In this scenario, when a user with a verified email at Clerk signs in using an OAuth provider that returns a matching but unverified email, Clerk will initiate a verification process. Once the email is verified, the OAuth account is linked to the existing one, and the user is signed in.

### When email address is not verified in Clerk and not in OAuth

> In instances where Clerk allows account creation without email verification at sign-up, there are scenarios where an account is created using an unverified email, via OAuth or other methods (e.g. username/password).

Reviewer: ```suggestion
For instances that allow account creation without email verification at sign-up, there is a possibility that an account is created using an unverified email, via OAuth or other methods (e.g. username/password).
```

<Callout type="info">

>   To create such instances you need to uncheck the "Verify at sign-up" toggle in the Clerk Dashboard.

Reviewer: ```suggestion
  To allow unverified email addresses for your instance you need to uncheck the "Verify at sign-up" toggle in the Clerk Dashboard.
```
